### PR TITLE
aks-preview: add azure-keyvault-secrets-provider as an aks addon

### PR DESCRIPTION
**Pull request by ZeroMagic**
This change extends the az aks cli commands to allow enabling/disabling the `azure-keyvault-secrets-provider` AKS addon as a part of `az aks create`, `az aks enable-addons` and `az aks disable-addons` commands.

Currently an optional `enableSecretRotation` can be specified with the feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Review by miwithro — APPROVED** [review]

**Comment by miwithro** [comment]
#sign-off

**Comment by yonzhan** [comment]
aks-preview

**Comment by miwithro** [comment]
#sign-off

File: src/aks-preview/azext_aks_preview/custom.py [hunk 1]
```diff
@@ -2929,6 +2940,16 @@ def _update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
                 addon_profile = ManagedClusterAddonProfile(enabled=True, config={CONST_ACC_SGX_QUOTE_HELPER_ENABLED: "false"})
                 if enable_sgxquotehelper:
                     addon_profile.config[CONST_ACC_SGX_QUOTE_HELPER_ENABLED] = "true"
+            elif addon == CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME:
+                if addon_profile.enabled:
+                    raise CLIError('The azure-keyvault-secrets-provider addon is already enabled for this managed cluster.\n'
```
Comment (robbiezhang) [inline src/aks-preview/azext_aks_preview/custom.py:2945]: if the addon is enabled, can the user change the "enable-secret-rotation" flag value?

Comment (ZeroMagic) [reply src/aks-preview/azext_aks_preview/custom.py:2945]: Discussed with Michael, we should support the scenario. I will add it later.

Comment (ZeroMagic) [reply src/aks-preview/azext_aks_preview/custom.py:2968]: Done. PTAL @robbiezhang

**Review by fengzhou-msft — APPROVED** [review]

File: src/aks-preview/azext_aks_preview/custom.py [hunk 2]
```diff
@@ -1645,6 +1654,17 @@ def aks_update(cmd,     # pylint: disable=too-many-statements,too-many-branches,
...
 
+    azure_keyvault_secrets_provider_addon_profile = instance.addon_profiles.get(CONST_AZURE_KEYVAULT_SECRETS_PROVIDER_ADDON_NAME, None)
+    if enable_secret_rotation:
+        if azure_keyvault_secrets_provider_addon_profile is None or not azure_keyvault_secrets_provider_addon_profile.enabled:
+            raise CLIError('--enable_secret_rotation can only be specified when azure-keyvault-secrets-provider is enabled')
+        azure_keyvault_secrets_provider_addon_profile.config[CONST_SECRET_ROTATION_ENABLED] = "true"
+
+    if disable_secret_rotation:
+        if azure_keyvault_secrets_provider_addon_profile is None or not azure_keyvault_secrets_provider_addon_profile.enabled:
+            raise CLIError('--enable_secret_rotation can only be specified when azure-keyvault-secrets-provider is enabled')
```
Comment (zqingqing1) [inline src/aks-preview/azext_aks_preview/custom.py:1665]: typo: --disable_secret_rotation

**Review by zqingqing1 — CHANGES_REQUESTED** [review]
you should update release note at history.md and bump version at setup.py.

File: src/aks-preview/azext_aks_preview/custom.py (hunk 2 above)
Comment (ZeroMagic) [reply src/aks-preview/azext_aks_preview/custom.py]: done

**Review by zqingqing1 — APPROVED** [review]
lgtm

**Pull request merged**